AllReadyApp/Web-App/AllReady/wwwroot/js/activity.js

```diff
@@ -1,57 +1,118 @@
 ﻿///<reference path="../lib/jquery/dist/jquery.js" />
 ///<reference path="../lib/knockout/dist/knockout.js" />
-(function (ko, $, tasks, skills, userSkills, isVolunteeredForActivity) {
+(function (ko, $, tasks, skills, userSkills, isVolunteeredForActivity, signupModelSeed) {
 
     function RequiredSkill(userSkills, skill)
     {
         var obj = ko.utils.extend({}, skill);
-        obj.IsUserSkill = userSkills && userSkills.length && userSkills.filter(function (userSkill) { return obj.Id === userSkill.Id }).length > 0;
+        obj.IsUserSkill = ko.observable(userSkills && userSkills.length && userSkills.filter(function (userSkill) { return obj.Id === userSkill.Id }).length > 0);
         return obj;
     }
 
-    function ActivityViewModel(tasks, skills, userSkills) {
-        this.skills = skills.map(RequiredSkill.bind(null, userSkills));
-        this.unassociatedSkills = this.skills.filter(function (skill) { return !skill.IsUserSkill; });
+    function ActivityViewModel(tasks, skills, userSkills, signupModel) {
+        var self = this;
+        self.skills = ko.observableArray(skills.map(RequiredSkill.bind(null, userSkills)));
+        self.unassociatedSkills = ko.computed(function () { return self.skills().filter(function (skill) { return !skill.IsUserSkill(); }) }); 
         tasks = tasks.map(function (task) { task.NotCompleteReason = ko.observable(); return task; });
-        this.tasks = ko.observableArray(tasks).filterBeforeDate("EndDateTime").textFilter(["Name", "Description"]);
-        this.enrolled = ko.observable(isVolunteeredForActivity);
-        this.errorUnenrolling = ko.observable(false);
-    }
+        self.tasks = ko.observableArray(tasks).filterBeforeDate("EndDateTime").textFilter(["Name", "Description"]);
+        self.enrolled = ko.observable(isVolunteeredForActivity);
+        self.errorUnenrolling = ko.observable(false);
 
-    var activityViewModel;
+        self.signupForActivity = function() {
+             HTBox.showModal({
+                 viewModel: new SignupViewModel(signupModel, self.unassociatedSkills),
+                 template: "VolunteerModal",
+                 context: self
+             }).then(activitySignupSuccess);
+        };
 
-    ActivityViewModel.prototype = {
-        enroll: function (activity) {
-            // TODO: set up a spinner
-            $.ajax({
-                type: "POST",
-                url: '/api/activity/' + activity + '/signup',
-                contentType: "application/json",
-            }).then(function (data) {
-                activityViewModel.enrolled(true);
-            }).fail(function (fail) {
-                console.log(fail);
+        function activitySignupSuccess(signUpViewModel) {
+            self.skills().forEach(function(skill) {
+                if (signUpViewModel.AddSkillIds().indexOf(skill.Id) >= 0){ skill.IsUserSkill(true) }

```

Comment:
I can't find any evidence in the code where `AddSkillIds` becomes an observable array. I think it's just a plain array. Additionally, you should put the curly braces on separate lines and put a semicolon after `skill.IsUserSkill(true)`.